app/models/track.rb

```diff
@@ -43,6 +86,70 @@ def to_param
     short_name
   end
 
+  def transition_possible?(transition)
+    self.class.state_machine.events_for(current_state).include?(transition)
+  end
+
+  # Gives the role of the track_organizer to the submitter
+  def assign_role_to_submitter
+    submitter.add_role 'track_organizer', self
+  end
+
+  # Revokes the track organizer role and removes the track from events that have it set
+  def revoke_role_and_cleanup
+    role = Role.find_by(name: 'track_organizer', resource: self)
+
+    if role
+      role.users.each do |user|
+        user.remove_role 'track_organizer', self
+      end
+    end
+
+    events.each do |event|
+      event.track = nil
+    end
+  end
+
+  ##
+  # Checks if the track is accepted
+  # ====Returns
+  # * +true+ -> If the track's state is 'accepted'
+  # * +false+ -> If the track's state isn't 'accepted'
+  def accepted?
+    state == 'accepted'
+  end
+
+  ##
+  # Checks if the track is confirmed
+  # ====Returns
+  # * +true+ -> If the track's state is 'confirmed'
+  # * +false+ -> If the track's state isn't 'confirmed'
+  def confirmed?
+    state == 'confirmed'
+  end
+
+  ##
+  # Checks if a self-organized track is accepted or confirmed
+  # ====Returns
+  # * +true+ -> If the track's state is 'accepted' or 'confirmed'
+  # * +false+ -> If the track's state is neither 'accepted' nor 'confirmed'
+  def self_organized_and_accepted_or_confirmed?
+    self_organized? && (accepted? || confirmed?)
+  end
+
+  def update_state(transition)
+    error = ''
+
+    begin
+      send(transition)
+    rescue Transitions::InvalidTransition => e
+      error += "State update failed. #{e.message} "
```

Comment:
Perhaps the error message can be a bit more informative regarding which event and/or which state the function was trying to change.